nestkernel/conn_builder_impl.h

```diff
@@ -0,0 +1,243 @@
+        sgid, &target, target_thread, synapse_model_ );
+    else if ( default_weight_ )
+      kernel().connection_manager.connect( sgid,
+        &target,
+        target_thread,
+        synapse_model_,
+        delay_->value_double( target_thread, rng ) );
+    else
+    {
+      double delay = delay_->value_double( target_thread, rng );
+      double weight = weight_->value_double( target_thread, rng );
+      kernel().connection_manager.connect(
+        sgid, &target, target_thread, synapse_model_, delay, weight );
+    }
+  }
+  else
+  {
+    assert( kernel().vp_manager.get_num_threads() == param_dicts_.size() );
+
+    for ( ConnParameterMap::const_iterator it = synapse_params_.begin();
+          it != synapse_params_.end();
+          ++it )
+    {
+      if ( it->first == names::receptor_type
+        || it->first == names::music_channel
+        || it->first == names::synapse_label )
+      {
+        try
+        {
+          // change value of dictionary entry without allocating new datum
+          IntegerDatum* id = static_cast< IntegerDatum* >(
+            ( ( *param_dicts_[ target_thread ] )[ it->first ] ).datum() );
+          ( *id ) = it->second->value_int( target_thread, rng );
+        }
+        catch ( KernelException& e )
+        {
+          if ( it->first == names::receptor_type )
+          {
+            throw BadProperty( "Receptor type must be of type integer." );
+          }
+          else if ( it->first == names::music_channel )
+          {
+            throw BadProperty( "Music channel type must be of type integer." );
+          }
+          else if ( it->first == names::synapse_label )
+          {
+            throw BadProperty( "Synapse label must be of type integer." );
+          }
+        }
+      }
+      else
+      {
+        // change value of dictionary entry without allocating new datum
+        DoubleDatum* dd = static_cast< DoubleDatum* >(
+          ( ( *param_dicts_[ target_thread ] )[ it->first ] ).datum() );
+        ( *dd ) = it->second->value_double( target_thread, rng );
+      }
+    }
+
+    if ( default_weight_and_delay_ )
+      kernel().connection_manager.connect( sgid,
+        &target,
+        target_thread,
+        synapse_model_,
+        param_dicts_[ target_thread ] );
+    else if ( default_weight_ )
+      kernel().connection_manager.connect( sgid,
+        &target,
+        target_thread,
+        synapse_model_,
+        param_dicts_[ target_thread ],
+        delay_->value_double( target_thread, rng ) );
+    else
+    {
+      double delay = delay_->value_double( target_thread, rng );
+      double weight = weight_->value_double( target_thread, rng );
+      kernel().connection_manager.connect( sgid,
+        &target,
+        target_thread,
+        synapse_model_,
+        param_dicts_[ target_thread ],
+        delay,
+        weight );
+    }
+  }
+}
+
+inline void
+ConnBuilder::skip_conn_parameter_( thread target_thread )
```

Comment:
This one is probably borderline for inlining. I would generally think that anything that contains loops should probably not be inlined, except, maybe if the loop would be empty (`begin()==end()`) in most cases. That might actually be the case here.